examples/extensible-interface-files/core-plugins/basic-example/Example.hs

```diff
@@ -0,0 +1,245 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Example where
+
+import Bag
+import GHC.Plugins
+import GHC.Iface.Load hiding (loadCore)
+import GHC.Iface.Syntax
+import GHC.IfaceToCore hiding (tcIfaceModGuts)
+import GHC.Tc.Types
+import GHC.Tc.Utils.Monad
+import GHC.Hs.Binds
+import GHC.Hs.Extension
+import GHC.Core.ConLike
+import Maybes as M
+import GHC.Iface.Env
+import GHC.Iface.Binary
+import GHC.Core.InstEnv
+import GHC.Core.FamInstEnv
+import GHC.CoreToIface
+
+
+-- We use the `interfaceWriteAction` to access the core bindings that are in the
+-- `ModGuts` as a result of the whole core pipeline. We then use serialise those
+-- bindings into the interface file using `registerInterfaceDataWith` to produce
+-- an extensible interface field from a plugin.
+
+-- We use the `typeCheckResultAction` to intercept the start of the core pipeline
+-- and perform our inlining step before any of the other core steps have run,
+-- including GHC's normal optimisation passes.
+plugin :: Plugin
+plugin = defaultPlugin { typeCheckResultAction = corePlugin
+                       , interfaceWriteAction  = interfacePlugin
+                       }
+
+-- Here we perform the serialisation of the `mg_binds` field from the `ModGuts`.

```

Comment:
Add: "The `mg_binds` contain ..., which we want to use in `typeCheckResultAction` to inline ...